lua/autorun/server/sv_limit_explosives.lua

```diff
@@ -0,0 +1,52 @@
+local limits = {
+    m9k_proxy = 4
+}
+
+local function getExplosivesTable(ply, class)
+    if limits[class] == nil then return end
+    
+    ply.placedExplosives = ply.placedExplosives or {}
+    ply.placedExplosives[class] = ply.placedExplosives[class] or {}
+    
+    return ply.placedExplosives[class]
+end
+
+local function onExplosiveCreated( owner, ent )
+    if not IsValid( owner ) then return end
+    if not IsValid( ent ) then return end
+
+    local class = ent:GetClass()
+    local explosives = getExplosivesTable( owner, class )
+    if explosives == nil then return end
+    
+    table.insert( explosives, 1, ent )
+    
+    if #explosives > limits[class] then
+        explosives[#explosives]:Remove()
+        explosives[#explosives] = nil
+    end
+
+end
+
+local function onExplosiveRemoved( owner, ent )
+    if not IsValid( owner ) then return end
+    if not IsValid( ent ) then return end
+    
+    local explosives = getExplosivesTable(owner, ent:GetClass())
```

Comment:
I don't think `getExplosiveTable` will return an empty table here because if an explosive is being removed it will have been created (and added to the table) at some point point. if it does `table.RemoveByValue` can handle empty tables.